tests/test_mathematical.py

```diff
@@ -24,3 +24,61 @@ def test_diff(array):
     result = inp.linalg.det(ia)
     expected = numpy.linalg.det(a)
     numpy.testing.assert_allclose(expected, result)
+
+
+class TestNanCumSum:
+
+    def test_nancumsum_dim1_int(self):
+        a = numpy.array([1, 2, 3, 4, 5])
+        ia = dpnp.array(a)
+
+        result = dpnp.nancumsum(ia)
+        expected = numpy.nancumsum(a)
+        numpy.testing.assert_array_equal(expected, result)
+
+
+    def test_nancumsum_dim1_nan(self):
+        a = numpy.array([1, 2, numpy.nan, 4, 5])
+        ia = dpnp.array(a)
+
+        result = dpnp.nancumsum(ia)
+        expected = numpy.nancumsum(a)
+        numpy.testing.assert_array_equal(expected, result)
+
+
+    def test_nancumsum_dim2_nan(self):
+        a = numpy.array([[1, 2, numpy.nan], [3, -4, -5]])
+        ia = dpnp.array(a)
+
+        result = dpnp.nancumsum(ia)
+        expected = numpy.nancumsum(a)
+        numpy.testing.assert_array_equal(expected, result)
+
+
+class TestNanCumProd:
+
+    def test_nancumprod_dim1_int(self):
```

Comment:
These 3 tests are the same but uses diffrent input data. Maybe it is better to use `pytest` features to combine tests into one with input variations. Example:
https://github.com/IntelPython/dpnp/blob/66b880c0483c3e803eca9edbbb6a07da27fc3020/tests/third_party/cupy/fft_tests/test_fft.py#L39-L62